core/src/main/java/bio/terra/pearl/core/service/survey/SurveyResponseService.java

```diff
@@ -209,24 +203,37 @@ protected SurveyResponse findOrCreateResponse(ParticipantTask task, Enrollee enr
     protected ParticipantTask updateTaskToResponse(ParticipantTask task, SurveyResponse response,
                                                    List<Answer> updatedAnswers, DataAuditInfo auditInfo) {
         task.setSurveyResponseId(response.getId());
-        if (task.getStatus() != TaskStatus.COMPLETE) { // task statuses shouldn't ever change from complete to not
-            if (response.isComplete()) {
-
-                if (task.getTaskType().equals(TaskType.CONSENT)) {
-                    // consent tasks are only marked as complete if they consented
-                    boolean isConsented = isConsented(response, answerService.findByResponse(response.getId()));
-                    task.setStatus(isConsented ? TaskStatus.COMPLETE : TaskStatus.REJECTED);
-                } else {
-                    task.setStatus(TaskStatus.COMPLETE);
-                }
-            } else if (task.getStatus() == TaskStatus.NEW && updatedAnswers.size() == 0) {
-                // if the task is new and no answers we submitted, this is just indicating the survey was viewed
-                task.setStatus(TaskStatus.VIEWED);
-            } else if (updatedAnswers.size() > 0) {
-                task.setStatus(TaskStatus.IN_PROGRESS);
+        TaskStatus updatedStatus = computeNewStatus(task, response, updatedAnswers);
+        if (task.getStatus() != updatedStatus || task.getSurveyResponseId() != response.getId()) {
+            task.setStatus(updatedStatus);
+            task.setSurveyResponseId(response.getId());
+            task = participantTaskService.update(task, auditInfo);
+        }
+        return task;
+    }
+
+    protected static TaskStatus computeNewStatus(ParticipantTask task, SurveyResponse response, List<Answer> updatedAnswers) {
+        if (task.getStatus() == TaskStatus.COMPLETE) {
```

Comment:
I appreciate the robustness of this, but I do find it strange that I was able to complete a survey, clear required responses and it still be marked as completed.

Reply:
Yeah, that's a good point about deleting past answers.  "Completed" is much less about "has all the data" and much more "Participant has at some point indicated that it's ready"